### ISSUE TITLE
Regression in `UriExtensions`: `inmemory:/` URI scheme is not supported any more

### ISSUE THREAD

**Issue by kittaakos** [eclipse/xtext-core#456]
After merging GH-403 into the master branch, `inmemory:/` URI scheme cannot be used for testing as those will be converted into `file:///` URI scheme.

Test case:
```java
@Test
def void testInMemory_GH_403() {
  initialize
  val uri = 'inmemory:/foo/bar.testlang';
  open(uri, '''
    type Foo { }
  ''');
  assertNotNull('''Diagnostics could be empty but never null.''', diagnostics.get(uri));
}
```

Additional details are [here](https://github.com/eclipse/xtext-core/pull/403).

**Comment by kittaakos** [comment]
The question is whether one has to use `AbstractLanguageServerTest.getVirtualFile(String)` instead of the `inmemory:/` scheme in the tests or this regression has to be fixed?

CC: @svenefftinge, @akosyakov

**Comment by cdietrich** [comment]
dont we have such tests in xtext-core?

**Comment by kittaakos** [comment]
The closest I have found was this one: https://github.com/eclipse/xtext-core/blob/d0ef36771ac3f81eb1f673933c5cc95e60d640c5/org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/OpenDocumentTest.xtend#L97

But the `#getVirtualFile(String)` method will return with a URI with the `file:///` scheme. Maybe I have overlooked something and we have a better test.

**Comment by akosyakov** [comment]
We should fix it and add tests into `UriExtensionsTest`.

**Comment by kittaakos** [comment]
If this is accepted as a bug, I start working on the fix. Just give me the green light.

**Comment by kittaakos** [comment]
I will take care of it.

**Comment by kittaakos** [comment]
PR: #457.

**Comment by kittaakos** [comment]
Closed via #458.

**Comment by kittaakos** [comment]
Closed via #458.
